### PR TITLE
WIP: Vue 2.0 compatibility

### PR DESCRIPTION
**Pull request by Darkside73**
Migration to Vue 2 (unfinished):
- [x] use `vue` components
- [x] avoid prop mutation in toast component: emit `destroyed` event instead
- [x] replace `coerce` in `options` prop of toast to computed property
- [x] use proper lifecycle callbacks and apply other technical migrations
- [ ] fix `_moveToast` method
- [ ] finish webpack configuration according to new dependencies

Also `forceDestroy` option added to lazy destroy closeable toasts as well

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,17 +3,22 @@
   "version": "2.0.3",
   "description": "toasts for vuejs",
   "main": "./dist/vue-toast.min.js",
-  "dependencies": {},
+  "dependencies": {
```
Comment (AStaroverov) [inline package.json]: `dependencies` should be empty

File: package.json [hunk 2]
```diff
@@ -3,17 +3,22 @@
...
+    "vue": "^2.0.1",
+    "vue-loader": "^9.5.1"
+  },
   "devDependencies": {
     "autoprefixer": "^6.1.0",
     "babel-core": "^5.8.33",
     "babel-loader": "^5.3.3",
     "css-loader": "^0.22.0",
     "extract-text-webpack-plugin": "^0.9.1",
+    "node-sass": "^3.10.1",
```
Comment (AStaroverov) [inline package.json:13]: used sass it good idea 👍

File: src/manager.vue [hunk 3]
```diff
@@ -28,16 +37,19 @@ export default {
   methods: {
     // Public
     showToast(message, options) {
+      options = Object.assign(this.options, options || {})
```
Comment (AStaroverov) [inline src/manager.vue:40]: I think we should use here 

```
if (isObject(option)) {
   options = Object.assign({}, this.options, option)
}
```

it help dont change this.options, for this we have setOptions function.
`isObject` we can add it in `utils.js` file

File: src/manager.vue [hunk 4]
```diff
@@ -28,16 +37,19 @@ export default {
...
       this._moveToast()
 
       return this
     },
     setOptions(options) {
       this.options = Object.assign(this.options, options || {})
-
       return this
     },
+    onDestroyed (toast) {
```
Comment (AStaroverov) [inline src/manager.vue]: add `_` for name, its like more private method

**Review by AStaroverov — COMMENTED** [review]
Nice 👍 Thx for u)

**Comment by vanderlin** [comment]
any chance this will get pulled in?

**Comment by Darkside73** [comment]
I've ended with this: https://gist.github.com/Darkside73/cbaa64b9e6eff687c709edb083be8bc9

**Comment by jancel** [comment]
Is there a chance of getting this merged?

**Comment by AStaroverov** [comment]
When all points are completed

**Comment by AStaroverov** [comment]
Now vue-toasts compatibility with Vue 2.1.8

**Pull request closed without merging**